### PR TITLE
Add stream.baseProperties + cross-platform SpeckleSettings paths

### PR DESCRIPTION
**Pull request by tsvilans**
### Add stream.baseProperties

- Add `StreamBaseProperties` schema
- Add optional `baseProperties` property to `Stream` schema
- Solves #23 

### Cross-platform paths

- Add preliminary support for cross-platform `SpeckleSettings` path.

### Other

- Bumped required `pydantic` version to 1.4.

### CODE REVIEW

File: speckle/Cache.py [hunk 1]
```diff
@@ -42,14 +42,29 @@ def __init__(self, filepath=None):
...
         if filepath is None:
-            self.db_path = os.path.join(os.getenv('LOCALAPPDATA'), os.path.join('SpeckleSettings', 'SpeckleCache.db'))
+
+            """
+            Get platform-dependent default SpeckleSettings directory
+            """
+            settings_dir = ""
+            if platform.system() == "Windows":
+                settings_dir = os.path.join(os.getenv('LOCALAPPDATA'), 'SpeckleSettings')
+            elif platform.system() == "Darwin":
```
Comment (AntoineDao) [inline speckle/Cache.py:52]: I would use `else` here as any other system is going to be Unix based and therefore use the "Home" env var

**Review by AntoineDao — APPROVED** [review]
Made a small proposal but other than that looks good to me :smile:

File: speckle/Cache.py (hunk 1 above)
Comment (tsvilans) [reply speckle/Cache.py:52]: Cool! I have very little experience with variables on other platforms :)

**Pull request merged**